# Final Project Ideas 1: Transportation to Work

The American Community Survey (ACS) includes questions on all kinds
of demographic, social, and economic data. Through various groups, we
can get at the data sliced and diced in all kinds of different ways.

In this notebook, we will discover that there are many groups that have
to do with how people get to work. We'll look briefly at some of this data
and then propose a number of possible project you could choose to do to 
use this data along with some of the tools you learned about in the lessons
and practiced in the exercises.

## Background: ACS Data on Transportation to Work

In [1]:
import censusdis.data as ced
from censusdis.datasets import ACS5

import pandas as pd

pd.set_option("max_colwidth", 500)
pd.set_option("display.max_rows", 100)

### What Groups Deal with Transportation to Work?

As we are about to see, there are a lot of different groups that combine data
on how people get to work with other data.

In [2]:
df_transportation_to_work_groups = ced.variables.search_groups(
    ACS5, 2022, pattern="transportation to work", case=False
)

In [3]:
df_transportation_to_work_groups

DATASET  YEAR    GROUP  \
0   acs/acs5  2022   B08006   
1   acs/acs5  2022   B08101   
2   acs/acs5  2022   B08103   
3   acs/acs5  2022  B08105A   
4   acs/acs5  2022  B08105B   
5   acs/acs5  2022  B08105C   
6   acs/acs5  2022  B08105D   
7   acs/acs5  2022  B08105E   
8   acs/acs5  2022  B08105F   
9   acs/acs5  2022  B08105G   
10  acs/acs5  2022  B08105H   
11  acs/acs5  2022  B08105I   
12  acs/acs5  2022   B08111   
13  acs/acs5  2022   B08113   
14  acs/acs5  2022   B08119   
15  acs/acs5  2022   B08121   
16  acs/acs5  2022   B08122   
17  acs/acs5  2022   B08124   
18  acs/acs5  2022   B08126   
19  acs/acs5  2022   B08128   
20  acs/acs5  2022   B08130   
21  acs/acs5  2022   B08132   
22  acs/acs5  2022   B08134   
23  acs/acs5  2022   B08136   
24  acs/acs5  2022   B08137   
25  acs/acs5  2022   B08141   
26  acs/acs5  2022   B08301   
27  acs/acs5  2022   B08406   
28  acs/acs5  2022   B08501   
29  acs/acs5  2022   B08503   
30  acs/acs5  2022  B08505A   
31  acs/acs5  2022  B08505B   
32  acs/acs5  2022  B08505C   
33  acs/acs5  2022  B08505D   
34  acs/acs5  2022  B08505E   
35  acs/acs5  2022  B08505F   
36  acs/acs5  2022  B08505G   
37  acs/acs5  2022  B08505H   
38  acs/acs5  2022  B08505I   
39  acs/acs5  2022   B08511   
40  acs/acs5  2022   B08513   
41  acs/acs5  2022   B08519   
42  acs/acs5  2022   B08521   
43  acs/acs5  2022   B08522   
44  acs/acs5  2022   B08524   
45  acs/acs5  2022   B08526   
46  acs/acs5  2022   B08528   
47  acs/acs5  2022   B08532   
48  acs/acs5  2022   B08534   
49  acs/acs5  2022   B08536   
50  acs/acs5  2022   B08537   
51  acs/acs5  2022   B08541   
52  acs/acs5  2022   B08601   
53  acs/acs5  2022   B99080   
54  acs/acs5  2022   B99086   

                                                                                                                                DESCRIPTION  
0                                                                                         Sex of Workers by Means of Transportation to Work  
1                                                                                                    Means of Transportation to Work by Age  
2                                                                                             Median Age by Means of Transportation to Work  
3                                                                                             Means of Transportation to Work (White Alone)  
4                                                                         Means of Transportation to Work (Black or African American Alone)  
5                                                                 Means of Transportation to Work (American Indian and Alaska Native Alone)  
6                                                                                             Means of Transportation to Work (Asian Alone)  
7                                                        Means of Transportation to Work (Native Hawaiian and Other Pacific Islander Alone)  
8                                                                                   Means of Transportation to Work (Some Other Race Alone)  
9                                                                                       Means of Transportation to Work (Two or More Races)  
10                                                                    Means of Transportation to Work (White Alone, Not Hispanic or Latino)  
11                                                                                     Means of Transportation to Work (Hispanic or Latino)  
12                                                                                    Means of Transportation to Work by Citizenship Status  
13                                                  Means of Transportation to Work by Language Spoken at Home and Ability to Speak English  
14                          Means of Transportation to Work by Workers' Earnings in the Past 12 Months (in 2022 Inflation-Adjusted Dollars

### A Slight Diversion on "for Workplace Geography"

You may have noticed in the groups above that there are often pairs of groups with similar
names, but one ends in "for Workplace Geography." This means that when the data is aggregated
by geography, such as by state, county, etc..., those groups aggregate the data by where 
people work. The groups without this suffix on their names aggregate it by where people live.

Let's look at an example to make this clear. First, compare the two groups `B08301` and `B08601`.
They measure the same thing, but the former aggregates by where people live and the latter by
where they work.

In [4]:
df_transportation_to_work_groups[
    df_transportation_to_work_groups["GROUP"].isin(["B08301", "B08601"])
]

DATASET  YEAR   GROUP  \
26  acs/acs5  2022  B08301   
52  acs/acs5  2022  B08601   

                                                DESCRIPTION  
26                          Means of Transportation to Work  
52  Means of Transportation to Work for Workplace Geography

The two groups have essentially identical variables except they are aggregated differently.

In [5]:
ced.variables.search(ACS5, 2022, group_name="B08301")

YEAR   DATASET   GROUP     VARIABLE  \
0   2022  acs/acs5  B08301  B08301_001E   
1   2022  acs/acs5  B08301  B08301_002E   
2   2022  acs/acs5  B08301  B08301_003E   
3   2022  acs/acs5  B08301  B08301_004E   
4   2022  acs/acs5  B08301  B08301_005E   
5   2022  acs/acs5  B08301  B08301_006E   
6   2022  acs/acs5  B08301  B08301_007E   
7   2022  acs/acs5  B08301  B08301_008E   
8   2022  acs/acs5  B08301  B08301_009E   
9   2022  acs/acs5  B08301  B08301_010E   
10  2022  acs/acs5  B08301  B08301_011E   
11  2022  acs/acs5  B08301  B08301_012E   
12  2022  acs/acs5  B08301  B08301_013E   
13  2022  acs/acs5  B08301  B08301_014E   
14  2022  acs/acs5  B08301  B08301_015E   
15  2022  acs/acs5  B08301  B08301_016E   
16  2022  acs/acs5  B08301  B08301_017E   
17  2022  acs/acs5  B08301  B08301_018E   
18  2022  acs/acs5  B08301  B08301_019E   
19  2022  acs/acs5  B08301  B08301_020E   
20  2022  acs/acs5  B08301  B08301_021E   
21  2022  acs/acs5  B08301       GEO_ID   
22  2022  acs/acs5  B08301         NAME   

                                                                                                                            LABEL  \
0                                                                                                                Estimate!!Total:   
1                                                                                           Estimate!!Total:!!Car, truck, or van:   
2                                                                              Estimate!!Total:!!Car, truck, or van:!!Drove alone   
3                                                                               Estimate!!Total:!!Car, truck, or van:!!Carpooled:   
4                                                          Estimate!!Total:!!Car, truck, or van:!!Carpooled:!!In 2-person carpool   
5                                                          Estimate!!Total:!!Car, truck, or van:!!Carpooled:!!In 3-person carpool   
6                                                          Estimate!!Total:!!Car, truck, or van:!!Carpooled:!!In 4-person carpool   
7                                                    Estimate!!Total:!!Car, truck, or van:!!Carpooled:!!In 5- or 6-person carpool   
8                                                  Estimate!!Total:!!Car, truck, or van:!!Carpooled:!!In 7-or-more-person carpool   
9                                                                    Estimate!!Total:!!Public transportation (excluding taxicab):   
10                                                              Estimate!!Total:!!Public transportation (excluding taxicab):!!Bus   
11                                          Estimate!!Total:!!Public transportation (excluding taxicab):!!Subway or elevated rail   
12                             Estimate!!Total:!!Public transportation (excluding taxicab):!!Long-distance train or commuter rail   
13  Estimate!!Total:!!Public transportation (excluding taxicab):!!Light rail, streetcar or trolley (carro público in Puerto Rico)   
14                                                        Estimate!!Total:!!Public transportation (excluding taxicab):!!Ferryboat   
15                                                                                                      Estimate!!Total:!!Taxicab   
16                                                                                                   Estimate!!Total:!!Motorcycle   
17                                                                                                      Estimate!!Total:!!Bicycle   
18                                                                                                       Estimate!!Total:!!Walked   
19                                                                                                  Estimate!!Total:!!Other means   
20                                                                                             Estimate!!Total:!!Worked from home   
21                                           

In [6]:
ced.variables.search(ACS5, 2022, group_name="B08601")

YEAR   DATASET   GROUP     VARIABLE  \
0   2022  acs/acs5  B08601  B08601_001E   
1   2022  acs/acs5  B08601  B08601_002E   
2   2022  acs/acs5  B08601  B08601_003E   
3   2022  acs/acs5  B08601  B08601_004E   
4   2022  acs/acs5  B08601  B08601_005E   
5   2022  acs/acs5  B08601  B08601_006E   
6   2022  acs/acs5  B08601  B08601_007E   
7   2022  acs/acs5  B08601  B08601_008E   
8   2022  acs/acs5  B08601  B08601_009E   
9   2022  acs/acs5  B08601  B08601_010E   
10  2022  acs/acs5  B08601  B08601_011E   
11  2022  acs/acs5  B08601  B08601_012E   
12  2022  acs/acs5  B08601  B08601_013E   
13  2022  acs/acs5  B08601  B08601_014E   
14  2022  acs/acs5  B08601  B08601_015E   
15  2022  acs/acs5  B08601  B08601_016E   
16  2022  acs/acs5  B08601  B08601_017E   
17  2022  acs/acs5  B08601  B08601_018E   
18  2022  acs/acs5  B08601  B08601_019E   
19  2022  acs/acs5  B08601  B08601_020E   
20  2022  acs/acs5  B08601  B08601_021E   
21  2022  acs/acs5  B08601       GEO_ID   
22  2022  acs/acs5  B08601         NAME   

                                                                                                                            LABEL  \
0                                                                                                                Estimate!!Total:   
1                                                                                           Estimate!!Total:!!Car, truck, or van:   
2                                                                              Estimate!!Total:!!Car, truck, or van:!!Drove alone   
3                                                                               Estimate!!Total:!!Car, truck, or van:!!Carpooled:   
4                                                          Estimate!!Total:!!Car, truck, or van:!!Carpooled:!!In 2-person carpool   
5                                                          Estimate!!Total:!!Car, truck, or van:!!Carpooled:!!In 3-person carpool   
6                                                          Estimate!!Total:!!Car, truck, or van:!!Carpooled:!!In 4-person carpool   
7                                                    Estimate!!Total:!!Car, truck, or van:!!Carpooled:!!In 5- or 6-person carpool   
8                                                  Estimate!!Total:!!Car, truck, or van:!!Carpooled:!!In 7-or-more-person carpool   
9                                                                    Estimate!!Total:!!Public transportation (excluding taxicab):   
10                                                              Estimate!!Total:!!Public transportation (excluding taxicab):!!Bus   
11                                          Estimate!!Total:!!Public transportation (excluding taxicab):!!Subway or elevated rail   
12                             Estimate!!Total:!!Public transportation (excluding taxicab):!!Long-distance train or commuter rail   
13  Estimate!!Total:!!Public transportation (excluding taxicab):!!Light rail, streetcar or trolley (carro público in Puerto Rico)   
14                                                        Estimate!!Total:!!Public transportation (excluding taxicab):!!Ferryboat   
15                                                                                                      Estimate!!Total:!!Taxicab   
16                                                                                                   Estimate!!Total:!!Motorcycle   
17                                                                                                      Estimate!!Total:!!Bicycle   
18                                                                                                       Estimate!!Total:!!Walked   
19                                                                                                  Estimate!!Total:!!Other means   
20                                                                                             Estimate!!Total:!!Worked from home   
21                                           

Now let's do a query.

In [7]:
from censusdis.states import NJ
from censusdis.places.new_jersey import NEWARK_CITY

In [8]:
ced.download(
    ACS5, 2020, ["NAME", "B08301_001E", "B08601_001E"], state=NJ, place=NEWARK_CITY
)

STATE  PLACE                     NAME  B08301_001E  B08601_001E
0    34  51000  Newark city, New Jersey       115068       163511

The `B08301_001E` column tells us that 115,068 people who live in Newark take some form of transportation to work. 
The `B08601_001E` column tells us that 163,511 people take some form of transportation to a workplace in Newark.

Some of these are the same people, and some are not. People may commute to or from Newark.

On the other hand, the people who work at home and live in Newark are exactly the same 
people who work at home and work in Newark, so the numbers should be the same.

In [9]:
ced.download(
    ACS5, 2020, ["NAME", "B08301_021E", "B08601_021E"], state=NJ, place=NEWARK_CITY
)

STATE  PLACE                     NAME  B08301_021E  B08601_021E
0    34  51000  Newark city, New Jersey         3053         3053

# Project Ideas

Using the groups of variables above, we can answer a lot of different questions.
Here are some that you might like to explore in your project. You can dig deep on 
one or tackle several related questions. They are in no particular order.

1. What is the rate of bicycle commuting by state?
   - For each state you will need to know how many people commute by bicycle and
     how many total commuters there are.
   - Plot the results on a map.
2. In (pick the state of your choice) is there a difference between the types of
   transportation used by low-income workers and high-income workers to get to work?
   - First, answer the question at the state level.
   - Second, answer the question at the county level.
3. In (pick the state or county of your choice), who is most likely to use public transit
   to get to work?
   - what age group is most likely to use public transit to get to work?
   - what gender is most likely to use public transit to get to work?
   - what race or ethnicity is most likely to use public transit to get to work?